### DISCUSSION TITLE
PorousFlow EnthalpySink depreciated?

### DISCUSSION THREAD

**Discussion by burritotukituki** [idaholab/moose#26888] [Q&A Modules: Porous Flow]
Hi all,

I've been wondering if EnthalpySink has been removed from PF. I need to run some simulations with complex well geometries, and in the past, I would rely on EnthalpySink for this purpose. However, it seems that it's no longer available.

Is there an alternative option? Perhaps PorousFlowLineSink? The issue with PorousFlowLineSink is that it lacks the T_in flag, so I can't fix the temperature I want to inject.

Does anyone have any suggestions or workarounds?

Thanks

**Comment by cpgr** [comment]
It's still there isn't it?

**Comment by burritotukituki** [comment]
Hi Chris,

Sorry I got confused. I remember that there was a PolyLineSink which could also had T_in as flag. Here an example with Falcon:

https://github.com/idaholab/falcon/blob/devel/examples/rtes_htm/Cranfield_HTM_continous_season_caps_water_newboundary_coarse.i

Where there is a formulation as
```

    type = FunctionEnthalpySink
    fluid_phase = 0
    variable = temperature
    SumQuantityUO = heat_enthalpy_in_inc
    line_base = '1 -9.99 -9.99 3.6'
    line_length = ${well_length}
    line_direction = '0 0 1'
    pressure = porepressure
    function = T_inj_function
    fp = tabulated_water
    p_or_t_vals = '-1e9 1e9'
    fluxes = '-${inj_ext_flux} -${inj_ext_flux}'

```
Should be in PorousFlow, right?

**Reply by cpgr** [reply]
Is PorousFlowEnthalpySink.C functionally equivalent?

**Reply by burritotukituki** [reply]
Hi Chris,
Well... is not quite the same... 

Looks like PorousFlowEnthalpySink is a boundary condition and not a DiracKernel.

I guess the difference is that I need to define it as a BC and therefore I need to have nodesets at the injection points, which complicates things a bit when the mesh is generated.

I can try to modify PorousFlowLineSink to include T_in to inject enthalpy at certain temperature.

Let me know if that worth to try :)

Cheers!
